### PR TITLE
Improve support for mem pools in TMOP

### PR DESCRIPTION
**Pull request by artv3**
- [ ] Remove changes in pmesh.{hpp/cpp}

### CODE REVIEW

File: mesh/pmesh.cpp [hunk 1]
```diff
@@ -2619,9 +2619,18 @@ void ParMesh::GetGhostFaceTransformation(
 }
 
 FaceElementTransformations *ParMesh::
-GetSharedFaceTransformations(int sf, bool fill2)
+GetSharedFaceTransformations(int sf, bool fill2, bool direct)
 {
-   int FaceNo = GetSharedFace(sf);
+   //AV: Disregard this change, just temporary while we do some refactoring
```
Comment (artv3) [inline mesh/pmesh.cpp:2624]: Please disregard this change, not part of this PR.

File: mesh/pmesh.hpp [hunk 2]
```diff
@@ -347,9 +347,9 @@ class ParMesh : public Mesh
 
    /** Get the FaceElementTransformations for the given shared face (edge 2D).
        In the returned object, 1 and 2 refer to the local and the neighbor
-       elements, respectively. */
+       elements, respectively. Use direct if sf is the face number */
    FaceElementTransformations *
-   GetSharedFaceTransformations(int sf, bool fill2 = true);
+   GetSharedFaceTransformations(int sf, bool fill2 = true, bool direct = false);
```
Comment (artv3) [inline mesh/pmesh.hpp:352]: Please disregard this change, not part of this PR.

**Comment by tomstitt** [comment]
I want to profile the memory usage again after we're up-to-date with _master_ and passing tests so this will probably change. I'm happy to wait or merge as-is

**Comment by artv3** [comment]
No longer needed.
